plugins/json_ld.ts

```diff
@@ -0,0 +1,102 @@
+import { merge } from "../core/utils/object.ts";
+import { getDataValue } from "../core/utils/data_values.ts";
+
+import type Site from "../core/site.ts";
+import type { Page } from "../core/file.ts";
+import { Graph, Thing } from "npm:schema-dts@1.1.2";
+export interface Options {
+  /** The list extensions this plugin applies to */
+  extensions?: string[];
+
+  /** The key name for the transformations definitions */
+  name?: string;
+}
+
+const defaults: Options = {
+  extensions: [".html"],
+  name: "json_ld",
+};
+export type JsonldData = Graph | Thing;
+
+// Helper function to determine if a value is an object
+function isObject(value: unknown) {
+  return value && typeof value === "object" && !Array.isArray(value);
+}
+
+/**
+ * A plugin to insert structured JSON-LD data for SEO and social media
+ * @see https://lume.land/plugins/json_ld/
+ */
+export function json_ld(userOptions?: Options) {
+  const options = merge(defaults, userOptions);
+
+  return (site: Site) => {
+    site.mergeKey(options.name, "object");
+    site.process(options.extensions, (pages) => pages.forEach(json_ld));
+
+    function json_ld(page: Page) {
```

Comment:
For clarity, use a different name for this function, because it's already used as the export function. Maybe `jsonLdProcessor` (camelCase).